doc/develop/test/babblesim.rst

```diff
@@ -0,0 +1,484 @@
+*************
+
+In usual case Pytest test runner is used most often for Python unit tests.
+However, thanks to the extensive system of hooks and plugins Pytest make it
+possible to define how tests gathering should looks like and what should be
+performed during test run. It enables to design own plugin dedicated for
+BabbleSim tests.
+
+Using Pytest test runner allows to use ready-made functionalities as:
+
+- scanning directories for tests' definitions
+- filtering tests by name (including/excluding tests)
+- tests execution and storing tests' results
+- generating several types of test report (``junit`` (``xml``), ``json``,
+  ``html``)
+- splitting tests into sub-groups (by ``split-tests`` plugin)
+- running tests in parallel (by ``xdist`` plugin)
+
+How to run tests?
+*****************
+
+To run BabbleSim tests defined to run by Pytest, following environmental
+variables have to be exported (of course below paths may vary depending on
+user's Zephyr and BabbleSim installation place):
+
+::
+
+    export ZEPHYR_BASE="${HOME}/zephyrproject/zephyr"
+    export BSIM_OUT_PATH="${HOME}/bsim/"
+    export BSIM_COMPONENTS_PATH="${HOME}/bsim/components/"
+
+Next below command should be called (from ZEPHYR_BASE directory level):
+
+::
+
+    pytest tests/bluetooth/bsim_bt
+
+Exemplary output could look like:
+
+::
+
+    ========================= test session starts ==========================
+    platform linux -- Python 3.8.10, pytest-6.2.4, py-1.10.0, pluggy-0.13.1
+    rootdir: /home/redbeard/zephyrproject/zephyr
+    plugins: typeguard-2.13.3, forked-1.4.0, xdist-2.5.0
+    collected 8 items
+
+    tests/bluetooth/bsim_bt/bsim_test_advx/bs_testsuite.yaml .        [ 12%]
+    tests/bluetooth/bsim_bt/bsim_test_app/bs_testsuite.yaml ....      [ 62%]
+    tests/bluetooth/bsim_bt/bsim_test_eatt/bs_testsuite.yaml ..       [ 87%]
+    tests/bluetooth/bsim_bt/bsim_test_gatt/bs_testsuite.yaml .        [100%]
+
+    ===================== 8 passed in 67.68s (0:01:07) =====================
+
+For more verbose output special arguments could be added to basic command:
+
+::
+
+    pytest tests/bluetooth/bsim_bt -vv -o log_cli=true
+
+How it works?
+*************
+
+BabbleSim tests are defined in ``bs_testsuite.yaml`` files (similar to "classic"
+Twister approach with defining tests in ``testcase.yaml`` files). At the
+beginning Pytest scan directories and try to find ``bs_testsuite.yaml`` and
+tests defined inside it. Exemplary ``bs_testsuite.yaml`` for
+``tests/bluetooth/bsim_bt/bsim_test_gatt`` test may look like below:
+
+.. code-block:: yaml
+
+    tests:
+      bluetooth.bsim.gatt:
+        platform_allow:
+          - nrf52_bsim
+        tags:
+          - bluetooth
+        extra_args:
+          - '-DCMAKE_C_FLAGS="-Werror"'
+          - "-DCONFIG_COVERAGE=y"
+          - "-DCMAKE_EXPORT_COMPILE_COMMANDS=ON"
+        bsim_config:
+          devices:
+            -
+              testid: gatt_client
+            -
+              testid: gatt_server
```

Comment:
I wanted to emphasis, that this is a list of devices (each with their own configuration like `testid` or `extra_run_args`), not the list of `testid`. I also considered to create list of devices, like this:
```yaml
devices:
  client:
    testid: gatt_client
  server:
    testid: gatt_server
```
But this approach requires adding specific name to each device which can be redundancy.

Generally I'm open for any propositions and if this style:
```yaml
devices:
  - testid: gatt_client
  - testid: gatt_server
```
is more readable for you I can change it ;)

Reply:
It may be that I am just not that used to using yaml files, but to me 

```yaml
devices:
  - testid: gatt_client
  - testid: gatt_server
```

looks better than

```yaml
devices:
  -
    testid: gatt_client
  -
    testid: gatt_server
```

How would the former look with `extra_run_args:` as well? 

Like this? 
```yaml
devices:
  - testid: gatt_client
    extra_run_args:
      - "-RealEncryption=0"
  - testid: gatt_server
```

Reply:
Yes, it would look like exactly as you wrote:
```yaml
devices:
  - testid: gatt_client
    extra_run_args:
      - "-RealEncryption=0"
  - testid: gatt_server
```